include/simics/dmllib.h

```diff
@@ -1428,6 +1428,38 @@ __qname(dml_qname_cache_t *cache, const char *fmt, ...)
         return (const char *)s;
 }
 
+UNUSED static const char *
+_DML_get_qname(_identity_t id, const _id_info_t *id_infos,
+               dml_qname_cache_t *cache) {
+    _id_info_t info = id_infos[id.id];
+
+    if (info.dimensions == 0) {
+        return info.logname;
+    }
+
+    uint32 *indices = MM_MALLOC(info.dimensions, uint32);
```

Comment:
why not a VLA?

Reply:
eh. I was a little paranoid of `info.dimensions` could get corrupted or something. There's no real reason.
Would you prefer a VLA?

Reply:
After some discussion on r.c., now a VLA.